runtime/src/main/java/org/corfudb/runtime/view/ManagementView.java

```diff
@@ -21,64 +22,65 @@
 @Slf4j
 public class ManagementView extends AbstractView {
 
-    private final long layoutRefreshTimeout = 500;
-
     public ManagementView(@NonNull CorfuRuntime runtime) {
         super(runtime);
     }
 
-    /**
-     * Checks if a given workflow with the given workflow ID is active or not.
-     * If returns True - the workflow is active
-     * If returns False - the workflow is either completed successfully or failed.
-     *
-     * @param server     Node endpoint on which the workflow is running.
-     * @param workflowId Workflow UUID
-     * @return True if active. Else False
-     */
-    public boolean isWorkflowActive(String server, UUID workflowId) {
-        return layoutHelper(l -> {
-            QueryResponse response = runtime.getRouter(server)
-                    .getClient(ManagementClient.class)
-                    .queryRequest(workflowId);
-            log.info("isWorkflowActive: {} on server:{} = {}",
-                workflowId, server, response.isActive());
-            return response.isActive();
-        });
+    void waitForWorkflow(UUID workflow, ManagementClient client) throws NetworkException,
+            TimeoutException {
+        long tries = runtime.getParameters().getWorkflowTimeout().getSeconds() /
+                runtime.getParameters().getWorkflowRetryRate().getSeconds();
+        for (long x = 0; x < tries; x++) {
+            if (!client.queryRequest(workflow).isActive()) {
+                return;
+            }
+            Sleep.sleepUninterruptibly(runtime.getParameters().getWorkflowRetryRate());
+            log.info("waitForWorkflow: waiting for {} on try {}", workflow, x);
+        }
+        log.debug("waitForWorkflow: workflow {} timeout", workflow);
+        throw new TimeoutException();
     }
 
     /**
      * Add a new node to the existing cluster.
      *
-     * @param endpoint Endpoint of the new node to be added to the cluster.
-     * @return True if completed successfully.
+     * @param endpointToAdd Endpoint of the new node to be added to the cluster.
+     * @throws WorkflowException throws a workflow exception on failure
      */
-    public boolean addNode(String endpoint) {
-        return layoutHelper(l -> {
+    public void addNode(String endpointToAdd) {
```

Comment:
addNode should take the timeouts as a parameter. Requiring a different timeout based on the workflow or caller seems like a common scenario.

Reply:
Yes.